gis/text_to_centreline_geometry_scripts/centreline_automation.sql

```diff
@@ -0,0 +1,1018 @@
+
+
+END;
+$geom$ LANGUAGE plpgsql;
+
+
+CREATE OR REPLACE FUNCTION crosic.match_line_to_centreline(line_geom geometry, highway2_before_editing text, metres_btwn1 FLOAT, metres_btwn2 FLOAT)
+RETURNS TABLE(geom geometry, street_name_arr TEXT[], ratio NUMERIC) AS $$
+DECLARE
+
+-- gardner expressway is called different things in the centreline intersection file and the centreline file
+-- so edit the name of roads that
+highway2 TEXT := CASE WHEN TRIM(highway2_before_editing) LIKE 'GARDINER EXPRESS%'
+	THEN 'F G Gardiner Xy W'
+	WHEN TRIM(highway2_before_editing) LIKE '%ALLEN RD%'
+	THEN 'William R Allen Rd S'
+	ELSE highway2_before_editing
+	END;
+
+street_name_arr TEXT[] := (SELECT ARRAY_AGG(lf_name)
+					 FROM
+					 (
+						SELECT DISTINCT lf_name, levenshtein(LOWER(highway2), LOWER(s.lf_name), 1, 1, 2) as lev_dist
+						FROM gis.centreline s
+						WHERE levenshtein(LOWER(highway2), LOWER(s.lf_name), 1, 1, 2) < 3
+						AND
+							(
+							(
+								-- "normal" case ... i.e. one intersection to another
+								-- metres_btwn and metres_btwn2 are null
+								COALESCE(metres_btwn1, metres_btwn2) IS NULL AND
+								-- 10* used to be 3
+								ST_DWithin( ST_Transform(s.geom, 26917), ST_BUFFER(line_geom, 3*ST_LENGTH(line_geom), 'endcap=flat join=round') , 10)
+								-- over 90% of centreline segment must be in the buffer
+								AND ST_Length(st_intersection(ST_BUFFER(line_geom, 3*(ST_LENGTH(line_geom)), 'endcap=flat join=round') , ST_Transform(s.geom, 26917))) /ST_Length(ST_Transform(s.geom, 26917)) > 0.9
+							)
+							OR
+							(
+								-- 10 TIMES LENGTH WORKS .... LOOK INTO POTENTIALLY CHANGING LATER
+								COALESCE(metres_btwn1, metres_btwn2) IS NOT NULL AND
+								ST_DWithin(ST_Transform(s.geom, 26917), ST_BUFFER(line_geom, 10*COALESCE(metres_btwn1, metres_btwn2), 'endcap=flat join=round'), 30)
+							)
+							)
+						ORDER BY levenshtein(LOWER(highway2), LOWER(s.lf_name), 1, 1, 2)
+						 ) AS x
+				);
+
+street_name TEXT = street_name_arr[1];
+
+geom geometry := (
+	SELECT ST_Transform(ST_UNION(s.geom), 26917) -- ST_LineMerge(ST_Transform(ST_UNION(s.geom), 26917))
+	FROM gis.centreline s
+	WHERE lf_name = street_name
+	AND
+	(
+	(
+		-- "normal" case ... i.e. one intersection to another
+		-- metres_btwn and metres_btwn2 are null
+		COALESCE(metres_btwn1, metres_btwn2) IS NULL AND
+		-- 3* could be changed to 10* if you would like
+		ST_DWithin( ST_Transform(s.geom, 26917), ST_BUFFER(line_geom, 3*ST_LENGTH(line_geom), 'endcap=flat join=round') , 10)
+		AND ST_Length(st_intersection(ST_BUFFER(line_geom, 3*(ST_LENGTH(line_geom)), 'endcap=flat join=round') , ST_Transform(s.geom, 26917))) /ST_Length(ST_Transform(s.geom, 26917)) > 0.9
+	)
+	OR
+	(
+		-- 10 TIMES LENGTH WORKS .... LOOK INTO POTENTIALLY CHANGING LATER
+		COALESCE(metres_btwn1, metres_btwn2) IS NOT NULL AND
+		ST_DWithin(ST_Transform(s.geom, 26917), ST_BUFFER(line_geom, 10*COALESCE(metres_btwn1, metres_btwn2), 'endcap=flat join=round'), 30)
```

Comment:
Use [`GREATEST`](https://devdocs.io/postgresql~9.6/functions-conditional#FUNCTIONS-GREATEST-LEAST) instead of COALESCE here